### PR TITLE
add target pool resource

### PR DESCRIPTION
**Pull request by gaohannk**
This is also for implement load balancer related image test

### CODE REVIEW

File: daisy/step_create_target_pool.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
+//  Copyright 2018 Google Inc. All Rights Reserved.
```
Comment (zoran15) [inline daisy/step_create_target_pool.go:1]: 2021

File: daisy/target_pool.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	"net/http"
+	"regexp"
+
+	daisyCompute "github.com/GoogleCloudPlatform/compute-image-tools/daisy/compute"
+	"google.golang.org/api/compute/v1"
+	"google.golang.org/api/googleapi"
+)
+
+var (
+	targetPoolURLRegex = regexp.MustCompile(fmt.Sprintf(`^(projects/(?P<project>%[1]s)/)?regions/(?P<zone>%[2]s)/TargetPools/(?P<targetPool>%[2]s)$`, projectRgxStr, rfc1035))
```
Comment (zoran15) [inline daisy/target_pool.go:16]: Isn't it `/targetPools/`? (lowercase)

File: daisy/target_pool.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	var errs DError
+	tp.Name, errs = tp.Resource.populateWithGlobal(ctx, s, tp.Name)
+
+	tp.Description = strOr(tp.Description, defaultDescription("TargetPool", s.w.Name, s.w.username))
+	tp.link = fmt.Sprintf("projects/%s/regions/%s/targetPools/%s", tp.Project, getRegionFromZone(s.w.Zone), tp.Name)
+	return errs
+}
+
+func (tp *TargetPool) validate(ctx context.Context, s *Step) DError {
+	pre := fmt.Sprintf("cannot create target-pool %q", tp.daisyName)
```
Comment (zoran15) [inline daisy/target_pool.go:46]: Use a consistent name for target pools (i.e. either target-pool, TargetPool, target pool, but not all of them)

File: daisy/target_pool.go (hunk 2 above)
Comment (gaohannk) [reply daisy/target_pool.go:16]: I thought the URL is case insensitive, but to make sure change to make it consistent with cloud document.

File: daisy/step_create_target_pool.go (hunk 1 above)
Comment (gaohannk) [reply daisy/step_create_target_pool.go:1]: Done

File: daisy/target_pool.go (hunk 3 above)
Comment (gaohannk) [reply daisy/target_pool.go:46]: Using` target pool` as plain text log and use `TargetPool` as ResourceTypeName when explicit indicate type

**Pull request closed without merging**